src/builder.rs

```diff
@@ -715,7 +714,11 @@ impl<'a, 'gcc, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'gcc, 'tcx> {
                 OperandValue::Ref(place.llval, Some(llextra), place.align)
             }
             else if place.layout.is_gcc_immediate() {
-                let load = self.load(place.llval.get_type(), place.llval, place.align);
+                let load = self.load(
+                    place.llval.get_type().get_pointee().expect("get_pointee"),
```

Comment:
One thing I'm wondering is whether we should get the type the same way it is done in the LLVM codegen, that is:
```suggestion
                    place.layout.llvm_type(self),
```

Reply:
Done. Had to pass an extra parameter (`set_fields`) to `gcc_type`.